Felienne/hedy
# Improve program reset on stop

**Description**

Improves reset when programs are stopped. When the stop button is clicked, all running timeout timers are cancelled. And when the user starts editing while a program is running, it is stopped and we return to the normal state.

**Fixes #820**

Always link the number of the issue or of the discussion that your pr concerns.

**How to test**

Samples taken from the issue ticket:
```
name is ask What is your name?
print hi, name
```
Execute the code, but when the question input field pops up, don't enter anything. Instead update the code, (stop and) restart the program, now do answer the question, and verify that the output is produced exactly once

Second example code:
```
print line 1
sleep 4
print line 2
sleep 4
print line 3
```
Run the program, stop and restart it (no changes needed) and verify that the output contains exactly 3 lines, in the correct order

**Checklist**
Done? Check if you have it all in place using this list:*
  

If you're unsure about any of these, don't hesitate to ask. We're here to help!
